hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/io/KeyInputStream.java

```diff
@@ -136,288 +155,41 @@ public static List<LengthInputStream> getStreamsFromKeyInfo(OmKeyInfo keyInfo,
     // Create a KeyInputStream for each part.
     for (Map.Entry<Integer, List<OmKeyLocationInfo>> entry :
         partsToBlocksMap.entrySet()) {
-      KeyInputStream keyInputStream = new KeyInputStream();
-      keyInputStream.initialize(keyInfo, entry.getValue(),
-          xceiverClientFactory, verifyChecksum, retryFunction,
+      List<BlockExtendedInputStream> streams = createStreams(keyInfo,
```

Comment:
```suggestion
      return partsToBlocksMap.values().stream().map(keyLocationInfos -> getFromOmKeyInfo(keyInfo, keyLocationInfos, xceiverClientFactory, verifyChecksum, retryFunction, blockStreamFactory)).collect(Collectors.toList());
```

Reply:
One need not compute length of each part. As far as I understand multipartOutputstream is doing this. No redundancy of logic required

Reply:
I simplified the multipart case, replaced the whole calculation with a groupingBy collector.